FlowAggregator tests fail on dual-stack clusters

**Describe the bug**
I observe very frequent failures of FlowAggregator e2e tests in CI for dual-stack K8s clusters. Actually, the dual-stack CI job has been failing consistently for a while, partly because of FlowAggregator test failures.

As an example, see the test logs below:
```
=== RUN   TestFlowAggregator/IPv4/LocalServiceAccess
    flowaggregator_test.go:560: Check the bandwidth using octetDeltaCountFromSourceNode in data record.
    flowaggregator_test.go:667: Iperf throughput: 11571.20 Mbits/s, IPFIX record throughput calculated through octetDeltaCountFromSourceNode: 11313.69 Mbits/s
    flowaggregator_test.go:569: Check the average bandwidth using octetTotalCountFromSourceNode 16949992005 in data record.
    flowaggregator_test.go:667: Iperf throughput: 11571.20 Mbits/s, IPFIX record throughput calculated through octetTotalCountFromSourceNode: 11299.99 Mbits/s
=== RUN   TestFlowAggregator/IPv4/RemoteServiceAccess
    flowaggregator_test.go:560: Check the bandwidth using octetDeltaCountFromSourceNode in data record.
    flowaggregator_test.go:667: Iperf throughput: 812.00 Mbits/s, IPFIX record throughput calculated through octetDeltaCountFromSourceNode: 829.10 Mbits/s
    flowaggregator_test.go:569: Check the average bandwidth using octetTotalCountFromSourceNode 1217975351 in data record.
    flowaggregator_test.go:667: Iperf throughput: 812.00 Mbits/s, IPFIX record throughput calculated through octetTotalCountFromSourceNode: 811.98 Mbits/s
=== RUN   TestFlowAggregator/IPv6
=== RUN   TestFlowAggregator/IPv6/IntraNodeFlows
    flowaggregator_test.go:750: Network Policies are realized.
    flowaggregator_test.go:560: Check the bandwidth using octetDeltaCountFromSourceNode in data record.
    flowaggregator_test.go:667: Iperf throughput: 30208.00 Mbits/s, IPFIX record throughput calculated through octetDeltaCountFromSourceNode: 6227702.58 Mbits/s
    flowaggregator_test.go:668: 
        	Error Trace:	flowaggregator_test.go:668
        	            				flowaggregator_test.go:562
        	            				flowaggregator_test.go:195
        	Error:      	Max difference between 6.2277025792e+06 and 30208 allowed is 4531.2, but difference was 6.1974945792e+06
        	Test:       	TestFlowAggregator/IPv6/IntraNodeFlows
        	Messages:   	Difference between Iperf bandwidth and IPFIX record bandwidth calculated through octetDeltaCountFromSourceNode should be lower than 15%
    flowaggregator_test.go:560: Check the bandwidth using octetDeltaCountFromSourceNode in data record.
    flowaggregator_test.go:667: Iperf throughput: 30208.00 Mbits/s, IPFIX record throughput calculated through octetDeltaCountFromSourceNode: 16336098454696.62 Mbits/s
    flowaggregator_test.go:668: 
        	Error Trace:	flowaggregator_test.go:668
        	            				flowaggregator_test.go:562
        	            				flowaggregator_test.go:195
        	Error:      	Max difference between 1.6336098454696625e+13 and 30208 allowed is 4531.2, but difference was 1.6336098424488625e+13
        	Test:       	TestFlowAggregator/IPv6/IntraNodeFlows
        	Messages:   	Difference between Iperf bandwidth and IPFIX record bandwidth calculated through octetDeltaCountFromSourceNode should be lower than 15%
    flowaggregator_test.go:560: Check the bandwidth using octetDeltaCountFromSourceNode in data record.
    flowaggregator_test.go:667: Iperf throughput: 30208.00 Mbits/s, IPFIX record throughput calculated through octetDeltaCountFromSourceNode: 34790577352189.41 Mbits/s
    flowaggregator_test.go:668: 
        	Error Trace:	flowaggregator_test.go:668
        	            				flowaggregator_test.go:562
        	            				flowaggregator_test.go:195
        	Error:      	Max difference between 3.479057735218941e+13 and 30208 allowed is 4531.2, but difference was 3.479057732198141e+13
        	Test:       	TestFlowAggregator/IPv6/IntraNodeFlows
        	Messages:   	Difference between Iperf bandwidth and IPFIX record bandwidth calculated through octetDeltaCountFromSourceNode should be lower than 15%
    flowaggregator_test.go:569: Check the average bandwidth using octetTotalCountFromSourceNode 1480569716211712 in data record.
    flowaggregator_test.go:667: Iperf throughput: 30208.00 Mbits/s, IPFIX record throughput calculated through octetTotalCountFromSourceNode: 987046477.47 Mbits/s
    flowaggregator_test.go:668: 
        	Error Trace:	flowaggregator_test.go:668
        	            				flowaggregator_test.go:570
        	            				flowaggregator_test.go:195
        	Error:      	Max difference between 9.870464774744745e+08 and 30208 allowed is 4531.2, but difference was 9.870162694744745e+08
        	Test:       	TestFlowAggregator/IPv6/IntraNodeFlows
        	Messages:   	Difference between Iperf bandwidth and IPFIX record bandwidth calculated through octetTotalCountFromSourceNode should be lower than 15%
=== RUN   TestFlowAggregator/IPv6/IntraNodeDenyConnIngressANP
I0616 14:58:39.445117   32529 k8s_util.go:623] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-anp-ingress-reject
I0616 14:58:39.463640   32529 k8s_util.go:623] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-anp-ingress-drop
    flowaggregator_test.go:832: Antrea Network Policies are realized.
=== RUN   TestFlowAggregator/IPv6/IntraNodeDenyConnEgressANP
I0616 14:58:44.147601   32529 k8s_util.go:623] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-anp-engress-reject
I0616 14:58:44.164011   32529 k8s_util.go:623] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-anp-engress-drop
    flowaggregator_test.go:832: Antrea Network Policies are realized.
=== RUN   TestFlowAggregator/IPv6/IntraNodeDenyConnNP
    flowaggregator_test.go:865: Network Policies are realized.
=== RUN   TestFlowAggregator/IPv6/InterNodeFlows
I0616 14:58:55.849810   32529 k8s_util.go:623] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-antrea-networkpolicy-ingress
I0616 14:58:55.868979   32529 k8s_util.go:623] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-antrea-networkpolicy-egress
    flowaggregator_test.go:785: Antrea Network Policies are realized.
    flowaggregator_test.go:560: Check the bandwidth using octetDeltaCountFromSourceNode in data record.
    flowaggregator_test.go:667: Iperf throughput: 830.00 Mbits/s, IPFIX record throughput calculated through octetDeltaCountFromSourceNode: 3727356160601.23 Mbits/s
    flowaggregator_test.go:668: 
        	Error Trace:	flowaggregator_test.go:668
        	            				flowaggregator_test.go:562
        	            				flowaggregator_test.go:321
        	Error:      	Max difference between 3.727356160601227e+12 and 830 allowed is 124.5, but difference was 3.727356159771227e+12
        	Test:       	TestFlowAggregator/IPv6/InterNodeFlows
        	Messages:   	Difference between Iperf bandwidth and IPFIX record bandwidth calculated through octetDeltaCountFromSourceNode should be lower than 15%
    flowaggregator_test.go:560: Check the bandwidth using octetDeltaCountFromSourceNode in data record.
    flowaggregator_test.go:667: Iperf throughput: 830.00 Mbits/s, IPFIX record throughput calculated through octetDeltaCountFromSourceNode: 6227702.58 Mbits/s
    flowaggregator_test.go:668: 
        	Error Trace:	flowaggregator_test.go:668
        	            				flowaggregator_test.go:562
        	            				flowaggregator_test.go:321
        	Error:      	Max difference between 6.2277025792e+06 and 830 allowed is 124.5, but difference was 6.2268725792e+06
        	Test:       	TestFlowAggregator/IPv6/InterNodeFlows
        	Messages:   	Difference between Iperf bandwidth and IPFIX record bandwidth calculated through octetDeltaCountFromSourceNode should be lower than 15%
    flowaggregator_test.go:560: Check the bandwidth using octetDeltaCountFromSourceNode in data record.
    flowaggregator_test.go:667: Iperf throughput: 830.00 Mbits/s, IPFIX record throughput calculated through octetDeltaCountFromSourceNode: 7438445141158.79 Mbits/s
    flowaggregator_test.go:668: 
        	Error Trace:	flowaggregator_test.go:668
        	            				flowaggregator_test.go:562
        	            				flowaggregator_test.go:321
        	Error:      	Max difference between 7.438445141158789e+12 and 830 allowed is 124.5, but difference was 7.438445140328789e+12
        	Test:       	TestFlowAggregator/IPv6/InterNodeFlows
        	Messages:   	Difference between Iperf bandwidth and IPFIX record bandwidth calculated through octetDeltaCountFromSourceNode should be lower than 15%
    flowaggregator_test.go:569: Check the average bandwidth using octetTotalCountFromSourceNode 30176440221696 in data record.
    flowaggregator_test.go:667: Iperf throughput: 830.00 Mbits/s, IPFIX record throughput calculated through octetTotalCountFromSourceNode: 20117626.81 Mbits/s
    flowaggregator_test.go:668: 
        	Error Trace:	flowaggregator_test.go:668
        	            				flowaggregator_test.go:570
        	            				flowaggregator_test.go:321
        	Error:      	Max difference between 2.0117626814464e+07 and 830 allowed is 124.5, but difference was 2.0116796814464e+07
        	Test:       	TestFlowAggregator/IPv6/InterNodeFlows
        	Messages:   	Difference between Iperf bandwidth and IPFIX record bandwidth calculated through octetTotalCountFromSourceNode should be lower than 15%
I0616 14:59:10.439709   32529 k8s_util.go:651] Deleting Antrea NetworkPolicy 'antrea-test/test-flow-aggregator-antrea-networkpolicy-ingress'
I0616 14:59:10.447578   32529 k8s_util.go:651] Deleting Antrea NetworkPolicy 'antrea-test/test-flow-aggregator-antrea-networkpolicy-egress'
=== RUN   TestFlowAggregator/IPv6/InterNodeDenyConnIngressANP
I0616 14:59:10.462729   32529 k8s_util.go:623] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-anp-ingress-reject
I0616 14:59:10.478263   32529 k8s_util.go:623] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-anp-ingress-drop
    flowaggregator_test.go:832: Antrea Network Policies are realized.
=== RUN   TestFlowAggregator/IPv6/InterNodeDenyConnEgressANP
I0616 14:59:15.197708   32529 k8s_util.go:623] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-anp-engress-reject
I0616 14:59:15.215853   32529 k8s_util.go:623] Creating/updating Antrea NetworkPolicy antrea-test/test-flow-aggregator-anp-engress-drop
    flowaggregator_test.go:832: Antrea Network Policies are realized.
    flowaggregator_test.go:673: 
        	Error Trace:	flowaggregator_test.go:673
        	            				flowaggregator_test.go:627
        	            				flowaggregator_test.go:391
        	Error:      	"\n  version: 10,  Message Length: 411\n  Exported Time: 1623855556 (2021-06-16 14:59:16 +0000 UTC)\n  Sequence No.: 643,  Observation Domain ID: 3273262736\nDATA SET:\n  DATA RECORD-0:\n    flowStartSeconds: 1623855550 \n    flowEndSeconds: 1623855550 \n    flowEndReason: 2 \n    sourceTransportPort: 43994 \n    destinationTransportPort: 5201 \n    protocolIdentifier: 6 \n    packetTotalCount: 1 \n    octetTotalCount: 80 \n    packetDeltaCount: 1 \n    octetDeltaCount: 80 \n    sourceIPv6Address: fd74:ca9b:172:16::c3a \n    destinationIPv6Address: fd74:ca9b:172:16:1::169d \n    reversePacketTotalCount: 0 \n    reverseOctetTotalCount: 0 \n    reversePacketDeltaCount: 0 \n    reverseOctetDeltaCount: 0 \n    sourcePodName:  \n    sourcePodNamespace:  \n    sourceNodeName:  \n    destinationPodName: perftest-c \n    destinationPodNamespace: antrea-test \n    destinationNodeName: antrea-ipv6-5-1 \n    destinationServicePort: 0 \n    destinationServicePortName:  \n    ingressNetworkPolicyName: test-flow-aggregator-anp-ingress-reject \n    ingressNetworkPolicyNamespace: antrea-test \n    ingressNetworkPolicyType: 2 \n    ingressNetworkPolicyRuleName: test-ingress-rule-name \n    ingressNetworkPolicyRuleAction: 3 \n    egressNetworkPolicyName:  \n    egressNetworkPolicyNamespace:  \n    egressNetworkPolicyType: 0 \n    egressNetworkPolicyRuleName:  \n    egressNetworkPolicyRuleAction: 0 \n    tcpState:  \n    flowType: 2 \n    destinationClusterIPv6: :: \n    originalExporterIPv6Address: ac10:101:10e6:5136::50 \n    originalObservationDomainId: 0 \n    octetDeltaCountFromSourceNode: 343597383680 \n    octetTotalCountFromSourceNode: 4294967296 \n    packetDeltaCountFromSourceNode: 4294967296 \n    packetTotalCountFromSourceNode: 0 \n    reverseOctetDeltaCountFromSourceNode: 0 \n    reverseOctetTotalCountFromSourceNode: 0 \n    reversePacketDeltaCountFromSourceNode: 0 \n    reversePacketTotalCountFromSourceNode: 343597383680 \n    octetDeltaCountFromDestinationNode: 343597383680 \n    octetTotalCountFromDestinationNode: 4294967296 \n    packetDeltaCountFromDestinationNode: 4294967296 \n    packetTotalCountFromDestinationNode: 0 \n    reverseOctetDeltaCountFromDestinationNode: 0 \n    reverseOctetTotalCountFromDestinationNode: 0 \nI0616 14:59:16.961082       1 collector.go:130] Processing IPFIX message\nI0616 14:59:16.961112       1 collector.go:130] Processing IPFIX message\nI0616 14:59:16.961264       1 collector.go:88] \n" does not contain "perftest-a"
        	Test:       	TestFlowAggregator/IPv6/InterNodeDenyConnEgressANP
        	Messages:   	Record with srcIP does not have Pod name
    flowaggregator_test.go:674: 
        	Error Trace:	flowaggregator_test.go:674
        	            				flowaggregator_test.go:627
        	            				flowaggregator_test.go:391
        	Error:      	"\n  version: 10,  Message Length: 411\n  Exported Time: 1623855556 (2021-06-16 14:59:16 +0000 UTC)\n  Sequence No.: 643,  Observation Domain ID: 3273262736\nDATA SET:\n  DATA RECORD-0:\n    flowStartSeconds: 1623855550 \n    flowEndSeconds: 1623855550 \n    flowEndReason: 2 \n    sourceTransportPort: 43994 \n    destinationTransportPort: 5201 \n    protocolIdentifier: 6 \n    packetTotalCount: 1 \n    octetTotalCount: 80 \n    packetDeltaCount: 1 \n    octetDeltaCount: 80 \n    sourceIPv6Address: fd74:ca9b:172:16::c3a \n    destinationIPv6Address: fd74:ca9b:172:16:1::169d \n    reversePacketTotalCount: 0 \n    reverseOctetTotalCount: 0 \n    reversePacketDeltaCount: 0 \n    reverseOctetDeltaCount: 0 \n    sourcePodName:  \n    sourcePodNamespace:  \n    sourceNodeName:  \n    destinationPodName: perftest-c \n    destinationPodNamespace: antrea-test \n    destinationNodeName: antrea-ipv6-5-1 \n    destinationServicePort: 0 \n    destinationServicePortName:  \n    ingressNetworkPolicyName: test-flow-aggregator-anp-ingress-reject \n    ingressNetworkPolicyNamespace: antrea-test \n    ingressNetworkPolicyType: 2 \n    ingressNetworkPolicyRuleName: test-ingress-rule-name \n    ingressNetworkPolicyRuleAction: 3 \n    egressNetworkPolicyName:  \n    egressNetworkPolicyNamespace:  \n    egressNetworkPolicyType: 0 \n    egressNetworkPolicyRuleName:  \n    egressNetworkPolicyRuleAction: 0 \n    tcpState:  \n    flowType: 2 \n    destinationClusterIPv6: :: \n    originalExporterIPv6Address: ac10:101:10e6:5136::50 \n    originalObservationDomainId: 0 \n    octetDeltaCountFromSourceNode: 343597383680 \n    octetTotalCountFromSourceNode: 4294967296 \n    packetDeltaCountFromSourceNode: 4294967296 \n    packetTotalCountFromSourceNode: 0 \n    reverseOctetDeltaCountFromSourceNode: 0 \n    reverseOctetTotalCountFromSourceNode: 0 \n    reversePacketDeltaCountFromSourceNode: 0 \n    reversePacketTotalCountFromSourceNode: 343597383680 \n    octetDeltaCountFromDestinationNode: 343597383680 \n    octetTotalCountFromDestinationNode: 4294967296 \n    packetDeltaCountFromDestinationNode: 4294967296 \n    packetTotalCountFromDestinationNode: 0 \n    reverseOctetDeltaCountFromDestinationNode: 0 \n    reverseOctetTotalCountFromDestinationNode: 0 \nI0616 14:59:16.961082       1 collector.go:130] Processing IPFIX message\nI0616 14:59:16.961112       1 collector.go:130] Processing IPFIX message\nI0616 14:59:16.961264       1 collector.go:88] \n" does not contain "sourcePodNamespace: antrea-test"
        	Test:       	TestFlowAggregator/IPv6/InterNodeDenyConnEgressANP
        	Messages:   	Record does not have correct sourcePodNamespace
    flowaggregator_test.go:675: 
        	Error Trace:	flowaggregator_test.go:675
        	            				flowaggregator_test.go:627
        	            				flowaggregator_test.go:391
        	Error:      	"\n  version: 10,  Message Length: 411\n  Exported Time: 1623855556 (2021-06-16 14:59:16 +0000 UTC)\n  Sequence No.: 643,  Observation Domain ID: 3273262736\nDATA SET:\n  DATA RECORD-0:\n    flowStartSeconds: 1623855550 \n    flowEndSeconds: 1623855550 \n    flowEndReason: 2 \n    sourceTransportPort: 43994 \n    destinationTransportPort: 5201 \n    protocolIdentifier: 6 \n    packetTotalCount: 1 \n    octetTotalCount: 80 \n    packetDeltaCount: 1 \n    octetDeltaCount: 80 \n    sourceIPv6Address: fd74:ca9b:172:16::c3a \n    destinationIPv6Address: fd74:ca9b:172:16:1::169d \n    reversePacketTotalCount: 0 \n    reverseOctetTotalCount: 0 \n    reversePacketDeltaCount: 0 \n    reverseOctetDeltaCount: 0 \n    sourcePodName:  \n    sourcePodNamespace:  \n    sourceNodeName:  \n    destinationPodName: perftest-c \n    destinationPodNamespace: antrea-test \n    destinationNodeName: antrea-ipv6-5-1 \n    destinationServicePort: 0 \n    destinationServicePortName:  \n    ingressNetworkPolicyName: test-flow-aggregator-anp-ingress-reject \n    ingressNetworkPolicyNamespace: antrea-test \n    ingressNetworkPolicyType: 2 \n    ingressNetworkPolicyRuleName: test-ingress-rule-name \n    ingressNetworkPolicyRuleAction: 3 \n    egressNetworkPolicyName:  \n    egressNetworkPolicyNamespace:  \n    egressNetworkPolicyType: 0 \n    egressNetworkPolicyRuleName:  \n    egressNetworkPolicyRuleAction: 0 \n    tcpState:  \n    flowType: 2 \n    destinationClusterIPv6: :: \n    originalExporterIPv6Address: ac10:101:10e6:5136::50 \n    originalObservationDomainId: 0 \n    octetDeltaCountFromSourceNode: 343597383680 \n    octetTotalCountFromSourceNode: 4294967296 \n    packetDeltaCountFromSourceNode: 4294967296 \n    packetTotalCountFromSourceNode: 0 \n    reverseOctetDeltaCountFromSourceNode: 0 \n    reverseOctetTotalCountFromSourceNode: 0 \n    reversePacketDeltaCountFromSourceNode: 0 \n    reversePacketTotalCountFromSourceNode: 343597383680 \n    octetDeltaCountFromDestinationNode: 343597383680 \n    octetTotalCountFromDestinationNode: 4294967296 \n    packetDeltaCountFromDestinationNode: 4294967296 \n    packetTotalCountFromDestinationNode: 0 \n    reverseOctetDeltaCountFromDestinationNode: 0 \n    reverseOctetTotalCountFromDestinationNode: 0 \nI0616 14:59:16.961082       1 collector.go:130] Processing IPFIX message\nI0616 14:59:16.961112       1 collector.go:130] Processing IPFIX message\nI0616 14:59:16.961264       1 collector.go:88] \n" does not contain "sourceNodeName: antrea-ipv6-5-0"
        	Test:       	TestFlowAggregator/IPv6/InterNodeDenyConnEgressANP
        	Messages:   	Record does not have correct sourceNodeName
=== RUN   TestFlowAggregator/IPv6/InterNodeDenyConnNP
    flowaggregator_test.go:865: Network Policies are realized.
    flowaggregator_test.go:676: 
        	Error Trace:	flowaggregator_test.go:676
        	            				flowaggregator_test.go:627
        	            				flowaggregator_test.go:426
        	Error:      	"\n  version: 10,  Message Length: 339\n  Exported Time: 1623855568 (2021-06-16 14:59:28 +0000 UTC)\n  Sequence No.: 678,  Observation Domain ID: 3273262736\nDATA SET:\n  DATA RECORD-0:\n    flowStartSeconds: 1623855562 \n    flowEndSeconds: 1623855563 \n    flowEndReason: 2 \n    sourceTransportPort: 53452 \n    destinationTransportPort: 5201 \n    protocolIdentifier: 6 \n    packetTotalCount: 2 \n    octetTotalCount: 160 \n    packetDeltaCount: 2 \n    octetDeltaCount: 160 \n    sourceIPv6Address: fd74:ca9b:172:16::c3b \n    destinationIPv6Address: fd74:ca9b:172:16:1::169e \n    reversePacketTotalCount: 0 \n    reverseOctetTotalCount: 0 \n    reversePacketDeltaCount: 0 \n    reverseOctetDeltaCount: 0 \n    sourcePodName: perftest-b \n    sourcePodNamespace: antrea-test \n    sourceNodeName: antrea-ipv6-5-0 \n    destinationPodName:  \n    destinationPodNamespace:  \n    destinationNodeName:  \n    destinationServicePort: 0 \n    destinationServicePortName:  \n    ingressNetworkPolicyName:  \n    ingressNetworkPolicyNamespace:  \n    ingressNetworkPolicyType: 0 \n    ingressNetworkPolicyRuleName:  \n    ingressNetworkPolicyRuleAction: 0 \n    egressNetworkPolicyName:  \n    egressNetworkPolicyNamespace:  \n    egressNetworkPolicyType: 1 \n    egressNetworkPolicyRuleName:  \n    egressNetworkPolicyRuleAction: 2 \n    tcpState:  \n    flowType: 2 \n    destinationClusterIPv6: :: \n    originalExporterIPv6Address: ac10:1:10e6:5137::a0 \n    originalObservationDomainId: 0 \n    octetDeltaCountFromSourceNode: 687194767360 \n    octetTotalCountFromSourceNode: 8589934592 \n    packetDeltaCountFromSourceNode: 8589934592 \n    packetTotalCountFromSourceNode: 0 \n    reverseOctetDeltaCountFromSourceNode: 0 \n    reverseOctetTotalCountFromSourceNode: 0 \n    reversePacketDeltaCountFromSourceNode: 0 \n    reversePacketTotalCountFromSourceNode: 687194767360 \n    octetDeltaCountFromDestinationNode: 687194767360 \n    octetTotalCountFromDestinationNode: 8589934592 \n    packetDeltaCountFromDestinationNode: 8589934592 \n    packetTotalCountFromDestinationNode: 0 \n    reverseOctetDeltaCountFromDestinationNode: 0 \n    reverseOctetTotalCountFromDestinationNode: 0 \nI0616 14:59:28.938344       1 collector.go:130] Processing IPFIX message\nI0616 14:59:28.938409       1 collector.go:88] \n" does not contain "perftest-e"
        	Test:       	TestFlowAggregator/IPv6/InterNodeDenyConnNP
        	Messages:   	Record with dstIP does not have Pod name
    flowaggregator_test.go:677: 
        	Error Trace:	flowaggregator_test.go:677
        	            				flowaggregator_test.go:627
        	            				flowaggregator_test.go:426
        	Error:      	"\n  version: 10,  Message Length: 339\n  Exported Time: 1623855568 (2021-06-16 14:59:28 +0000 UTC)\n  Sequence No.: 678,  Observation Domain ID: 3273262736\nDATA SET:\n  DATA RECORD-0:\n    flowStartSeconds: 1623855562 \n    flowEndSeconds: 1623855563 \n    flowEndReason: 2 \n    sourceTransportPort: 53452 \n    destinationTransportPort: 5201 \n    protocolIdentifier: 6 \n    packetTotalCount: 2 \n    octetTotalCount: 160 \n    packetDeltaCount: 2 \n    octetDeltaCount: 160 \n    sourceIPv6Address: fd74:ca9b:172:16::c3b \n    destinationIPv6Address: fd74:ca9b:172:16:1::169e \n    reversePacketTotalCount: 0 \n    reverseOctetTotalCount: 0 \n    reversePacketDeltaCount: 0 \n    reverseOctetDeltaCount: 0 \n    sourcePodName: perftest-b \n    sourcePodNamespace: antrea-test \n    sourceNodeName: antrea-ipv6-5-0 \n    destinationPodName:  \n    destinationPodNamespace:  \n    destinationNodeName:  \n    destinationServicePort: 0 \n    destinationServicePortName:  \n    ingressNetworkPolicyName:  \n    ingressNetworkPolicyNamespace:  \n    ingressNetworkPolicyType: 0 \n    ingressNetworkPolicyRuleName:  \n    ingressNetworkPolicyRuleAction: 0 \n    egressNetworkPolicyName:  \n    egressNetworkPolicyNamespace:  \n    egressNetworkPolicyType: 1 \n    egressNetworkPolicyRuleName:  \n    egressNetworkPolicyRuleAction: 2 \n    tcpState:  \n    flowType: 2 \n    destinationClusterIPv6: :: \n    originalExporterIPv6Address: ac10:1:10e6:5137::a0 \n    originalObservationDomainId: 0 \n    octetDeltaCountFromSourceNode: 687194767360 \n    octetTotalCountFromSourceNode: 8589934592 \n    packetDeltaCountFromSourceNode: 8589934592 \n    packetTotalCountFromSourceNode: 0 \n    reverseOctetDeltaCountFromSourceNode: 0 \n    reverseOctetTotalCountFromSourceNode: 0 \n    reversePacketDeltaCountFromSourceNode: 0 \n    reversePacketTotalCountFromSourceNode: 687194767360 \n    octetDeltaCountFromDestinationNode: 687194767360 \n    octetTotalCountFromDestinationNode: 8589934592 \n    packetDeltaCountFromDestinationNode: 8589934592 \n    packetTotalCountFromDestinationNode: 0 \n    reverseOctetDeltaCountFromDestinationNode: 0 \n    reverseOctetTotalCountFromDestinationNode: 0 \nI0616 14:59:28.938344       1 collector.go:130] Processing IPFIX message\nI0616 14:59:28.938409       1 collector.go:88] \n" does not contain "destinationPodNamespace: antrea-test"
        	Test:       	TestFlowAggregator/IPv6/InterNodeDenyConnNP
        	Messages:   	Record does not have correct destinationPodNamespace
    flowaggregator_test.go:678: 
        	Error Trace:	flowaggregator_test.go:678
        	            				flowaggregator_test.go:627
        	            				flowaggregator_test.go:426
        	Error:      	"\n  version: 10,  Message Length: 339\n  Exported Time: 1623855568 (2021-06-16 14:59:28 +0000 UTC)\n  Sequence No.: 678,  Observation Domain ID: 3273262736\nDATA SET:\n  DATA RECORD-0:\n    flowStartSeconds: 1623855562 \n    flowEndSeconds: 1623855563 \n    flowEndReason: 2 \n    sourceTransportPort: 53452 \n    destinationTransportPort: 5201 \n    protocolIdentifier: 6 \n    packetTotalCount: 2 \n    octetTotalCount: 160 \n    packetDeltaCount: 2 \n    octetDeltaCount: 160 \n    sourceIPv6Address: fd74:ca9b:172:16::c3b \n    destinationIPv6Address: fd74:ca9b:172:16:1::169e \n    reversePacketTotalCount: 0 \n    reverseOctetTotalCount: 0 \n    reversePacketDeltaCount: 0 \n    reverseOctetDeltaCount: 0 \n    sourcePodName: perftest-b \n    sourcePodNamespace: antrea-test \n    sourceNodeName: antrea-ipv6-5-0 \n    destinationPodName:  \n    destinationPodNamespace:  \n    destinationNodeName:  \n    destinationServicePort: 0 \n    destinationServicePortName:  \n    ingressNetworkPolicyName:  \n    ingressNetworkPolicyNamespace:  \n    ingressNetworkPolicyType: 0 \n    ingressNetworkPolicyRuleName:  \n    ingressNetworkPolicyRuleAction: 0 \n    egressNetworkPolicyName:  \n    egressNetworkPolicyNamespace:  \n    egressNetworkPolicyType: 1 \n    egressNetworkPolicyRuleName:  \n    egressNetworkPolicyRuleAction: 2 \n    tcpState:  \n    flowType: 2 \n    destinationClusterIPv6: :: \n    originalExporterIPv6Address: ac10:1:10e6:5137::a0 \n    originalObservationDomainId: 0 \n    octetDeltaCountFromSourceNode: 687194767360 \n    octetTotalCountFromSourceNode: 8589934592 \n    packetDeltaCountFromSourceNode: 8589934592 \n    packetTotalCountFromSourceNode: 0 \n    reverseOctetDeltaCountFromSourceNode: 0 \n    reverseOctetTotalCountFromSourceNode: 0 \n    reversePacketDeltaCountFromSourceNode: 0 \n    reversePacketTotalCountFromSourceNode: 687194767360 \n    octetDeltaCountFromDestinationNode: 687194767360 \n    octetTotalCountFromDestinationNode: 8589934592 \n    packetDeltaCountFromDestinationNode: 8589934592 \n    packetTotalCountFromDestinationNode: 0 \n    reverseOctetDeltaCountFromDestinationNode: 0 \n    reverseOctetTotalCountFromDestinationNode: 0 \nI0616 14:59:28.938344       1 collector.go:130] Processing IPFIX message\nI0616 14:59:28.938409       1 collector.go:88] \n" does not contain "destinationNodeName: antrea-ipv6-5-1"
        	Test:       	TestFlowAggregator/IPv6/InterNodeDenyConnNP
        	Messages:   	Record does not have correct destinationNodeName
=== RUN   TestFlowAggregator/IPv6/LocalServiceAccess
    flowaggregator_test.go:468: 
        	Error Trace:	flowaggregator_test.go:468
        	            				flowaggregator_test.go:436
        	Error:      	Received unexpected error:
        	            	strconv.ParseFloat: parsing "": invalid syntax
        	Test:       	TestFlowAggregator/IPv6/LocalServiceAccess
        	Messages:   	Error when converting iperf bandwidth  to float64 type
=== RUN   TestFlowAggregator/IPv6/RemoteServiceAccess
    flowaggregator_test.go:468: 
        	Error Trace:	flowaggregator_test.go:468
        	            				flowaggregator_test.go:446
        	Error:      	Received unexpected error:
        	            	strconv.ParseFloat: parsing "": invalid syntax
        	Test:       	TestFlowAggregator/IPv6/RemoteServiceAccess
        	Messages:   	Error when converting iperf bandwidth  to float64 type
=== CONT  TestFlowAggregator
    fixtures.go:338: Deleting 'flow-aggregator' K8s Namespace
    fixtures.go:224: Exporting test logs to '/var/lib/jenkins/workspace/antrea-ipv6-ds-e2e-for-pull-request/antrea-test-logs/TestFlowAggregator/beforeTeardown.Jun16-14-59-56'
    fixtures.go:328: Error when exporting kubelet logs: error when running journalctl on Node 'antrea-ipv6-5-0', is it available? Error: <nil>
    fixtures.go:349: Deleting 'antrea-test' K8s Namespace
--- FAIL: TestFlowAggregator (325.64s)
    --- PASS: TestFlowAggregator/IPv4 (100.00s)
        --- PASS: TestFlowAggregator/IPv4/IntraNodeFlows (15.78s)
        --- PASS: TestFlowAggregator/IPv4/IntraNodeDenyConnIngressANP (4.67s)
        --- PASS: TestFlowAggregator/IPv4/IntraNodeDenyConnEgressANP (4.58s)
        --- PASS: TestFlowAggregator/IPv4/IntraNodeDenyConnNP (6.45s)
        --- PASS: TestFlowAggregator/IPv4/InterNodeFlows (16.05s)
        --- PASS: TestFlowAggregator/IPv4/InterNodeDenyConnIngressANP (4.72s)
        --- PASS: TestFlowAggregator/IPv4/InterNodeDenyConnEgressANP (4.57s)
        --- PASS: TestFlowAggregator/IPv4/InterNodeDenyConnNP (10.03s)
        --- PASS: TestFlowAggregator/IPv4/LocalServiceAccess (15.71s)
        --- PASS: TestFlowAggregator/IPv4/RemoteServiceAccess (14.33s)
    --- FAIL: TestFlowAggregator/IPv6 (67.86s)
        --- FAIL: TestFlowAggregator/IPv6/IntraNodeFlows (14.36s)
        --- PASS: TestFlowAggregator/IPv6/IntraNodeDenyConnIngressANP (4.69s)
        --- PASS: TestFlowAggregator/IPv6/IntraNodeDenyConnEgressANP (4.86s)
        --- PASS: TestFlowAggregator/IPv6/IntraNodeDenyConnNP (6.85s)
        --- FAIL: TestFlowAggregator/IPv6/InterNodeFlows (14.61s)
        --- PASS: TestFlowAggregator/IPv6/InterNodeDenyConnIngressANP (4.74s)
        --- FAIL: TestFlowAggregator/IPv6/InterNodeDenyConnEgressANP (4.89s)
        --- FAIL: TestFlowAggregator/IPv6/InterNodeDenyConnNP (9.42s)
        --- FAIL: TestFlowAggregator/IPv6/LocalServiceAccess (0.15s)
        --- FAIL: TestFlowAggregator/IPv6/RemoteServiceAccess (0.16s)
```

It seems that the root cause is missing / empty Information Elements in the records sent by the aggregator to the collector. This is in turn causing record matching to fail, as well as bandwidth computation.

**Versions:**
N/A

--- a/ci/kind/test-e2e-kind.sh
+++ b/ci/kind/test-e2e-kind.sh
@@ -103,7 +103,7 @@ else
     manifest_args="$manifest_args --no-np"
 fi
 
-COMMON_IMAGES_LIST=("gcr.io/kubernetes-e2e-test-images/agnhost:2.8" "projects.registry.vmware.com/library/busybox" "projects.registry.vmware.com/antrea/nginx" "projects.registry.vmware.com/antrea/perftool" "projects.registry.vmware.com/antrea/ipfix-collector:v0.5.2")
+COMMON_IMAGES_LIST=("gcr.io/kubernetes-e2e-test-images/agnhost:2.8" "projects.registry.vmware.com/library/busybox" "projects.registry.vmware.com/antrea/nginx" "projects.registry.vmware.com/antrea/perftool" "projects.registry.vmware.com/antrea/ipfix-collector:v0.5.3")
 for image in "${COMMON_IMAGES_LIST[@]}"; do
     for i in `seq 3`; do
         docker pull $image && break

--- a/go.mod
+++ b/go.mod
@@ -42,7 +42,7 @@ require (
 	github.com/stretchr/testify v1.6.1
 	github.com/ti-mo/conntrack v0.3.0
 	github.com/vishvananda/netlink v1.1.0
-	github.com/vmware/go-ipfix v0.5.2
+	github.com/vmware/go-ipfix v0.5.3
 	golang.org/x/crypto v0.0.0-20210220033148-5ea612d1eb83
 	golang.org/x/exp v0.0.0-20200224162631-6cc2880d07d6
 	golang.org/x/mod v0.4.0

--- a/pkg/flowaggregator/flowaggregator.go
+++ b/pkg/flowaggregator/flowaggregator.go
@@ -16,7 +16,6 @@ package flowaggregator
 
 import (
 	"fmt"
-	"net"
 	"time"
 
 	"github.com/vmware/go-ipfix/pkg/collector"
@@ -177,8 +176,10 @@ type flowAggregator struct {
 	activeFlowRecordTimeout     time.Duration
 	inactiveFlowRecordTimeout   time.Duration
 	exportingProcess            ipfix.IPFIXExportingProcess
-	templateIDv4                uint16
-	templateIDv6                uint16
+	templateIDv4Expv4           uint16
+	templateIDv4Expv6           uint16
+	templateIDv6Expv4           uint16
+	templateIDv6Expv6           uint16
 	registry                    ipfix.IPFIXRegistry
 	set                         ipfixentities.Set
 	flowAggregatorAddress       string
@@ -222,10 +223,14 @@ func podInfoIndexFunc(obj interface{}) ([]string, error) {
 	if !ok {
 		return nil, fmt.Errorf("obj is not pod: %+v", obj)
 	}
-	if pod.Status.PodIP == "" {
-		klog.V(4).Infof("Pod %s/%s has not been assigned IP yet. ", pod.Namespace, pod.Name)
+	if len(pod.Status.PodIPs) > 0 && pod.Status.Phase != corev1.PodSucceeded && pod.Status.Phase != corev1.PodFailed {
+		indexes := make([]string, len(pod.Status.PodIPs))
+		for i := range pod.Status.PodIPs {
+			indexes[i] = pod.Status.PodIPs[i].IP
+		}
+		return indexes, nil
 	}
-	return []string{pod.Status.PodIP}, nil
+	return nil, nil
 }
 
 func (fa *flowAggregator) InitCollectingProcess() error {
@@ -294,6 +299,31 @@ func (fa *flowAggregator) InitAggregationProcess() error {
 	return err
 }
 
+func (fa *flowAggregator) createAndSendTemplate(isRecordIPv6, isOriginExporterIPv6 bool) (uint16, error) {
+	templateID := fa.exportingProcess.NewTemplateID()
+	// If Pod IPs (source and destination IP) in the flow record belong to IPv4 Family and
+	// original exporter IP belongs to IPv6 family, we will send template with ID templateIDv4Expv6,
+	// which has sourceIPv4Address, destinationIPv4Address and originalExporterIPv6Address.
+	// Same applies to other combinations.
+	recordIPFamily := "IPv4"
+	exporterIPFamily := "IPv4"
+	if isRecordIPv6 {
+		recordIPFamily = "IPv6"
+	}
+	if isOriginExporterIPv6 {
+		exporterIPFamily = "IPv6"
+	}
+	bytesSent, err := fa.sendTemplateSet(isRecordIPv6, isOriginExporterIPv6)
+	if err != nil {
+		fa.exportingProcess.CloseConnToCollector()
+		fa.exportingProcess = nil
+		fa.set.ResetSet()
+		return 0, fmt.Errorf("sending %s template set with %s original exporter ip failed, err: %v", recordIPFamily, exporterIPFamily, err)
+	}
+	klog.V(2).InfoS("Exporting process initialized", "bytesSent", bytesSent, "templateSetIPFamily", recordIPFamily, "originalExporterIPFamily", exporterIPFamily)
+	return templateID, nil
+}
+
 func (fa *flowAggregator) initExportingProcess() error {
 	// TODO: This code can be further simplified by changing the go-ipfix API to accept
 	// externalFlowCollectorAddr and externalFlowCollectorProto instead of net.Addr input.
@@ -324,29 +354,20 @@ func (fa *flowAggregator) initExportingProcess() error {
 		return fmt.Errorf("got error when initializing IPFIX exporting process: %v", err)
 	}
 	fa.exportingProcess = ep
-	// Currently, we send two templates for IPv4 and IPv6 regardless of the IP families supported by cluster
-	fa.templateIDv4 = fa.exportingProcess.NewTemplateID()
-	bytesSent, err := fa.sendTemplateSet(false)
-	if err != nil {
-		fa.exportingProcess.CloseConnToCollector()
-		fa.exportingProcess = nil
-		fa.templateIDv4 = 0
-		fa.set.ResetSet()
-		return fmt.Errorf("sending IPv4 template set failed, err: %v", err)
+	// Currently, we send 4 templates for covering all the cases in dual-stack clusters, where Pod IPs
+	// and original exporter IP could belong to different IP families.
+	if fa.templateIDv4Expv4, err = fa.createAndSendTemplate(false, false); err != nil {
+		return err
 	}
-	klog.V(2).Infof("Initialized exporting process and sent %d bytes size of IPv4 template set", bytesSent)
-
-	fa.templateIDv6 = fa.exportingProcess.NewTemplateID()
-	bytesSent, err = fa.sendTemplateSet(true)
-	if err != nil {
-		fa.exportingProcess.CloseConnToCollector()
-		fa.exportingProcess = nil
-		fa.templateIDv6 = 0
-		fa.set.ResetSet()
-		return fmt.Errorf("sending IPv6 template set failed, err: %v", err)
+	if fa.templateIDv4Expv6, err = fa.createAndSendTemplate(false, true); err != nil {
+		return err
 	}
-	klog.V(2).Infof("Initialized exporting process and sent %d bytes size of IPv6 template set", bytesSent)
-
+	if fa.templateIDv6Expv4, err = fa.createAndSendTemplate(true, false); err != nil {
+		return err
+	}
+	if fa.templateIDv6Expv6, err = fa.createAndSendTemplate(true, true); err != nil {
+		return err
+	}
 	return nil
 }
 
@@ -398,19 +419,31 @@ func (fa *flowAggregator) flowRecordExpiryCheck(stopCh <-chan struct{}) {
 	}
 }
 
-func (fa *flowAggregator) sendFlowKeyRecord(key ipfixintermediate.FlowKey, record ipfixintermediate.AggregationFlowRecord) error {
-	templateID := fa.templateIDv4
-	if net.ParseIP(key.SourceAddress).To4() == nil || net.ParseIP(key.DestinationAddress).To4() == nil {
-		templateID = fa.templateIDv6
+func (fa *flowAggregator) sendFlowKeyRecord(key ipfixintermediate.FlowKey, record *ipfixintermediate.AggregationFlowRecord) error {
+	isRecordIPv4 := fa.aggregationProcess.IsAggregatedRecordIPv4(*record)
+	isOriginExporterIPv4 := fa.aggregationProcess.IsExporterOfAggregatedRecordIPv4(*record)
+	var templateID uint16
+	if isRecordIPv4 {
+		if isOriginExporterIPv4 {
+			templateID = fa.templateIDv4Expv4
+		} else {
+			templateID = fa.templateIDv4Expv6
+		}
+	} else {
+		if isOriginExporterIPv4 {
+			templateID = fa.templateIDv6Expv4
+		} else {
+			templateID = fa.templateIDv6Expv6
+		}
 	}
 	// TODO: more records per data set will be supported when go-ipfix supports size check when adding records
 	fa.set.ResetSet()
 	if err := fa.set.PrepareSet(ipfixentities.Data, templateID); err != nil {
 		return err
 	}
-	if !fa.aggregationProcess.IsMetadataFilled(record) {
+	if !fa.aggregationProcess.AreCorrelatedFieldsFilled(*record) {
 		fa.fillK8sMetadata(key, record.Record)
-		fa.aggregationProcess.SetMetadataFilled(record)
+		fa.aggregationProcess.SetCorrelatedFieldsFilled(record)
 	}
 	err := fa.set.AddRecord(record.Record.GetOrderedElementList(), templateID)
 	if err != nil {
@@ -428,17 +461,25 @@ func (fa *flowAggregator) sendFlowKeyRecord(key ipfixintermediate.FlowKey, recor
 	return nil
 }
 
-func (fa *flowAggregator) sendTemplateSet(isIPv6 bool) (int, error) {
+func (fa *flowAggregator) sendTemplateSet(isFlowKeyIPv6 bool, isOriginalExporterIPv6 bool) (int, error) {
 	elements := make([]*ipfixentities.InfoElementWithValue, 0)
 	ianaInfoElements := ianaInfoElementsIPv4
 	antreaInfoElements := antreaInfoElementsIPv4
 	aggregatorElements := aggregatorElementsIPv4
-	templateID := fa.templateIDv4
-	if isIPv6 {
+	templateID := fa.templateIDv4Expv4
+	if isOriginalExporterIPv6 {
+		aggregatorElements = aggregatorElementsIPv6
+		templateID = fa.templateIDv4Expv6
+	}
+	if isFlowKeyIPv6 {
 		ianaInfoElements = ianaInfoElementsIPv6
 		antreaInfoElements = antreaInfoElementsIPv6
-		aggregatorElements = aggregatorElementsIPv6
-		templateID = fa.templateIDv6
+		if isOriginalExporterIPv6 {
+			aggregatorElements = aggregatorElementsIPv6
+			templateID = fa.templateIDv6Expv6
+		} else {
+			templateID = fa.templateIDv6Expv4
+		}
 	}
 	for _, ie := range ianaInfoElements {
 		element, err := fa.registry.GetInfoElement(ie, ipfixregistry.IANAEnterpriseID)

--- a/pkg/flowaggregator/flowaggregator_test.go
+++ b/pkg/flowaggregator/flowaggregator_test.go
@@ -29,8 +29,10 @@ import (
 )
 
 const (
-	testTemplateIDv4        = uint16(256)
-	testTemplateIDv6        = uint16(257)
+	testTemplateIDv4Expv4   = uint16(256)
+	testTemplateIDv4Expv6   = uint16(257)
+	testTemplateIDv6Expv4   = uint16(258)
+	testTemplateIDv6Expv6   = uint16(259)
 	testActiveTimeout       = 60 * time.Second
 	testInactiveTimeout     = 180 * time.Second
 	testObservationDomainID = 0xabcd
@@ -54,8 +56,10 @@ func TestFlowAggregator_sendFlowKeyRecord(t *testing.T) {
 		activeFlowRecordTimeout:     testActiveTimeout,
 		inactiveFlowRecordTimeout:   testInactiveTimeout,
 		exportingProcess:            mockIPFIXExpProc,
-		templateIDv4:                testTemplateIDv4,
-		templateIDv6:                testTemplateIDv6,
+		templateIDv4Expv4:           testTemplateIDv4Expv4,
+		templateIDv4Expv6:           testTemplateIDv4Expv6,
+		templateIDv6Expv4:           testTemplateIDv6Expv4,
+		templateIDv6Expv6:           testTemplateIDv6Expv6,
 		registry:                    mockIPFIXRegistry,
 		set:                         mockDataSet,
 		flowAggregatorAddress:       "",
@@ -77,7 +81,7 @@ func TestFlowAggregator_sendFlowKeyRecord(t *testing.T) {
 		DestinationPort:    5678,
 	}
 
-	readyRecord := ipfixintermediate.AggregationFlowRecord{
+	readyRecord := &ipfixintermediate.AggregationFlowRecord{
 		Record:      mockRecord,
 		ReadyToSend: true,
 	}
@@ -86,7 +90,7 @@ func TestFlowAggregator_sendFlowKeyRecord(t *testing.T) {
 		name       string
 		isIPv6     bool
 		flowKey    ipfixintermediate.FlowKey
-		flowRecord ipfixintermediate.AggregationFlowRecord
+		flowRecord *ipfixintermediate.AggregationFlowRecord
 	}{
 		{
 			"IPv4_ready_to_send",
@@ -103,9 +107,9 @@ func TestFlowAggregator_sendFlowKeyRecord(t *testing.T) {
 	}
 
 	for _, tc := range testcases {
-		templateID := fa.templateIDv4
+		templateID := fa.templateIDv4Expv4
 		if tc.isIPv6 {
-			templateID = fa.templateIDv6
+			templateID = fa.templateIDv6Expv6
 		}
 		mockDataSet.EXPECT().ResetSet()
 		mockDataSet.EXPECT().PrepareSet(ipfixentities.Data, templateID).Return(nil)
@@ -114,8 +118,10 @@ func TestFlowAggregator_sendFlowKeyRecord(t *testing.T) {
 		mockDataSet.EXPECT().AddRecord(elementList, templateID).Return(nil)
 		mockIPFIXExpProc.EXPECT().SendSet(mockDataSet).Return(0, nil)
 		mockAggregationProcess.EXPECT().ResetStatElementsInRecord(mockRecord).Return(nil)
-		mockAggregationProcess.EXPECT().IsMetadataFilled(tc.flowRecord).Return(false)
-		mockAggregationProcess.EXPECT().SetMetadataFilled(tc.flowRecord)
+		mockAggregationProcess.EXPECT().AreCorrelatedFieldsFilled(*tc.flowRecord).Return(false)
+		mockAggregationProcess.EXPECT().SetCorrelatedFieldsFilled(tc.flowRecord)
+		mockAggregationProcess.EXPECT().IsExporterOfAggregatedRecordIPv4(*tc.flowRecord).Return(!tc.isIPv6)
+		mockAggregationProcess.EXPECT().IsAggregatedRecordIPv4(*tc.flowRecord).Return(!tc.isIPv6)
 
 		err := fa.sendFlowKeyRecord(tc.flowKey, tc.flowRecord)
 		assert.NoError(t, err, "Error in sending flow key record: %v, key: %v, record: %v", err, tc.flowKey, tc.flowRecord)
@@ -138,8 +144,10 @@ func TestFlowAggregator_sendTemplateSet(t *testing.T) {
 		aggregationProcess:          nil,
 		activeFlowRecordTimeout:     testActiveTimeout,
 		exportingProcess:            mockIPFIXExpProc,
-		templateIDv4:                testTemplateIDv4,
-		templateIDv6:                testTemplateIDv6,
+		templateIDv4Expv4:           testTemplateIDv4Expv4,
+		templateIDv4Expv6:           testTemplateIDv4Expv6,
+		templateIDv6Expv4:           testTemplateIDv6Expv4,
+		templateIDv6Expv6:           testTemplateIDv6Expv6,
 		registry:                    mockIPFIXRegistry,
 		set:                         mockTempSet,
 		flowAggregatorAddress:       "",
@@ -151,12 +159,12 @@ func TestFlowAggregator_sendTemplateSet(t *testing.T) {
 		ianaInfoElements := ianaInfoElementsIPv4
 		antreaInfoElements := antreaInfoElementsIPv4
 		aggregatorElements := aggregatorElementsIPv4
-		testTemplateID := fa.templateIDv4
+		testTemplateID := fa.templateIDv4Expv4
 		if isIPv6 {
 			ianaInfoElements = ianaInfoElementsIPv6
 			antreaInfoElements = antreaInfoElementsIPv6
 			aggregatorElements = aggregatorElementsIPv6
-			testTemplateID = fa.templateIDv6
+			testTemplateID = fa.templateIDv6Expv6
 		}
 		// Following consists of all elements that are in ianaInfoElements and antreaInfoElements (globals)
 		// Only the element name is needed, other arguments have dummy values.
@@ -192,7 +200,7 @@ func TestFlowAggregator_sendTemplateSet(t *testing.T) {
 		// above elements: ianaInfoElements, ianaReverseInfoElements and antreaInfoElements.
 		mockIPFIXExpProc.EXPECT().SendSet(mockTempSet).Return(0, nil)
 
-		_, err := fa.sendTemplateSet(isIPv6)
+		_, err := fa.sendTemplateSet(isIPv6, isIPv6)
 		assert.NoErrorf(t, err, "Error in sending template record: %v, isIPv6: %v", err, isIPv6)
 	}
 }

--- a/pkg/ipfix/ipfix_intermediate.go
+++ b/pkg/ipfix/ipfix_intermediate.go
@@ -31,8 +31,10 @@ type IPFIXAggregationProcess interface {
 	ForAllExpiredFlowRecordsDo(callback ipfixintermediate.FlowKeyRecordMapCallBack) error
 	GetExpiryFromExpirePriorityQueue() time.Duration
 	ResetStatElementsInRecord(record ipfixentities.Record) error
-	SetMetadataFilled(record ipfixintermediate.AggregationFlowRecord)
-	IsMetadataFilled(record ipfixintermediate.AggregationFlowRecord) bool
+	SetCorrelatedFieldsFilled(record *ipfixintermediate.AggregationFlowRecord)
+	AreCorrelatedFieldsFilled(record ipfixintermediate.AggregationFlowRecord) bool
+	IsAggregatedRecordIPv4(record ipfixintermediate.AggregationFlowRecord) bool
+	IsExporterOfAggregatedRecordIPv4(record ipfixintermediate.AggregationFlowRecord) bool
 }
 
 type ipfixAggregationProcess struct {
@@ -71,10 +73,18 @@ func (ap *ipfixAggregationProcess) ResetStatElementsInRecord(record ipfixentitie
 	return ap.AggregationProcess.ResetStatElementsInRecord(record)
 }
 
-func (ap *ipfixAggregationProcess) SetMetadataFilled(record ipfixintermediate.AggregationFlowRecord) {
-	ap.AggregationProcess.SetMetadataFilled(record, true)
+func (ap *ipfixAggregationProcess) SetCorrelatedFieldsFilled(record *ipfixintermediate.AggregationFlowRecord) {
+	ap.AggregationProcess.SetCorrelatedFieldsFilled(record, true)
 }
 
-func (ap *ipfixAggregationProcess) IsMetadataFilled(record ipfixintermediate.AggregationFlowRecord) bool {
-	return ap.AggregationProcess.IsMetadataFilled(record)
+func (ap *ipfixAggregationProcess) AreCorrelatedFieldsFilled(record ipfixintermediate.AggregationFlowRecord) bool {
+	return ap.AggregationProcess.AreCorrelatedFieldsFilled(record)
+}
+
+func (ap *ipfixAggregationProcess) IsAggregatedRecordIPv4(record ipfixintermediate.AggregationFlowRecord) bool {
+	return ap.AggregationProcess.IsAggregatedRecordIPv4(record)
+}
+
+func (ap *ipfixAggregationProcess) IsExporterOfAggregatedRecordIPv4(record ipfixintermediate.AggregationFlowRecord) bool {
+	return ap.AggregationProcess.IsExporterOfAggregatedRecordIPv4(record)
 }

--- a/pkg/ipfix/testing/mock_ipfix.go
+++ b/pkg/ipfix/testing/mock_ipfix.go
@@ -225,6 +225,20 @@ func (m *MockIPFIXAggregationProcess) EXPECT() *MockIPFIXAggregationProcessMockR
 	return m.recorder
 }
 
+// AreCorrelatedFieldsFilled mocks base method
+func (m *MockIPFIXAggregationProcess) AreCorrelatedFieldsFilled(arg0 intermediate.AggregationFlowRecord) bool {
+	m.ctrl.T.Helper()
+	ret := m.ctrl.Call(m, "AreCorrelatedFieldsFilled", arg0)
+	ret0, _ := ret[0].(bool)
+	return ret0
+}
+
+// AreCorrelatedFieldsFilled indicates an expected call of AreCorrelatedFieldsFilled
+func (mr *MockIPFIXAggregationProcessMockRecorder) AreCorrelatedFieldsFilled(arg0 interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "AreCorrelatedFieldsFilled", reflect.TypeOf((*MockIPFIXAggregationProcess)(nil).AreCorrelatedFieldsFilled), arg0)
+}
+
 // ForAllExpiredFlowRecordsDo mocks base method
 func (m *MockIPFIXAggregationProcess) ForAllExpiredFlowRecordsDo(arg0 intermediate.FlowKeyRecordMapCallBack) error {
 	m.ctrl.T.Helper()
@@ -253,18 +267,32 @@ func (mr *MockIPFIXAggregationProcessMockRecorder) GetExpiryFromExpirePriorityQu
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "GetExpiryFromExpirePriorityQueue", reflect.TypeOf((*MockIPFIXAggregationProcess)(nil).GetExpiryFromExpirePriorityQueue))
 }
 
-// IsMetadataFilled mocks base method
-func (m *MockIPFIXAggregationProcess) IsMetadataFilled(arg0 intermediate.AggregationFlowRecord) bool {
+// IsAggregatedRecordIPv4 mocks base method
+func (m *MockIPFIXAggregationProcess) IsAggregatedRecordIPv4(arg0 intermediate.AggregationFlowRecord) bool {
 	m.ctrl.T.Helper()
-	ret := m.ctrl.Call(m, "IsMetadataFilled", arg0)
+	ret := m.ctrl.Call(m, "IsAggregatedRecordIPv4", arg0)
 	ret0, _ := ret[0].(bool)
 	return ret0
 }
 
-// IsMetadataFilled indicates an expected call of IsMetadataFilled
-func (mr *MockIPFIXAggregationProcessMockRecorder) IsMetadataFilled(arg0 interface{}) *gomock.Call {
+// IsAggregatedRecordIPv4 indicates an expected call of IsAggregatedRecordIPv4
+func (mr *MockIPFIXAggregationProcessMockRecorder) IsAggregatedRecordIPv4(arg0 interface{}) *gomock.Call {
 	mr.mock.ctrl.T.Helper()
-	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "IsMetadataFilled", reflect.TypeOf((*MockIPFIXAggregationProcess)(nil).IsMetadataFilled), arg0)
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "IsAggregatedRecordIPv4", reflect.TypeOf((*MockIPFIXAggregationProcess)(nil).IsAggregatedRecordIPv4), arg0)
+}
+
+// IsExporterOfAggregatedRecordIPv4 mocks base method
+func (m *MockIPFIXAggregationProcess) IsExporterOfAggregatedRecordIPv4(arg0 intermediate.AggregationFlowRecord) bool {
+	m.ctrl.T.Helper()
+	ret := m.ctrl.Call(m, "IsExporterOfAggregatedRecordIPv4", arg0)
+	ret0, _ := ret[0].(bool)
+	return ret0
+}
+
+// IsExporterOfAggregatedRecordIPv4 indicates an expected call of IsExporterOfAggregatedRecordIPv4
+func (mr *MockIPFIXAggregationProcessMockRecorder) IsExporterOfAggregatedRecordIPv4(arg0 interface{}) *gomock.Call {
+	mr.mock.ctrl.T.Helper()
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "IsExporterOfAggregatedRecordIPv4", reflect.TypeOf((*MockIPFIXAggregationProcess)(nil).IsExporterOfAggregatedRecordIPv4), arg0)
 }
 
 // ResetStatElementsInRecord mocks base method
@@ -281,16 +309,16 @@ func (mr *MockIPFIXAggregationProcessMockRecorder) ResetStatElementsInRecord(arg
 	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "ResetStatElementsInRecord", reflect.TypeOf((*MockIPFIXAggregationProcess)(nil).ResetStatElementsInRecord), arg0)
 }
 
-// SetMetadataFilled mocks base method
-func (m *MockIPFIXAggregationProcess) SetMetadataFilled(arg0 intermediate.AggregationFlowRecord) {
+// SetCorrelatedFieldsFilled mocks base method
+func (m *MockIPFIXAggregationProcess) SetCorrelatedFieldsFilled(arg0 *intermediate.AggregationFlowRecord) {
 	m.ctrl.T.Helper()
-	m.ctrl.Call(m, "SetMetadataFilled", arg0)
+	m.ctrl.Call(m, "SetCorrelatedFieldsFilled", arg0)
 }
 
-// SetMetadataFilled indicates an expected call of SetMetadataFilled
-func (mr *MockIPFIXAggregationProcessMockRecorder) SetMetadataFilled(arg0 interface{}) *gomock.Call {
+// SetCorrelatedFieldsFilled indicates an expected call of SetCorrelatedFieldsFilled
+func (mr *MockIPFIXAggregationProcessMockRecorder) SetCorrelatedFieldsFilled(arg0 interface{}) *gomock.Call {
 	mr.mock.ctrl.T.Helper()
-	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "SetMetadataFilled", reflect.TypeOf((*MockIPFIXAggregationProcess)(nil).SetMetadataFilled), arg0)
+	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "SetCorrelatedFieldsFilled", reflect.TypeOf((*MockIPFIXAggregationProcess)(nil).SetCorrelatedFieldsFilled), arg0)
 }
 
 // Start mocks base method

--- a/test/e2e/flowaggregator_test.go
+++ b/test/e2e/flowaggregator_test.go
@@ -16,6 +16,7 @@ package e2e
 
 import (
 	"fmt"
+	"net"
 	"regexp"
 	"strconv"
 	"strings"
@@ -108,8 +109,8 @@ const (
 	ingressDropANPName             = "test-flow-aggregator-anp-ingress-drop"
 	ingressDenyNPName              = "test-flow-aggregator-np-ingress-deny"
 	egressAllowNetworkPolicyName   = "test-flow-aggregator-networkpolicy-egress-allow"
-	egressRejectANPName            = "test-flow-aggregator-anp-engress-reject"
-	egressDropANPName              = "test-flow-aggregator-anp-engress-drop"
+	egressRejectANPName            = "test-flow-aggregator-anp-egress-reject"
+	egressDropANPName              = "test-flow-aggregator-anp-egress-drop"
 	egressDenyNPName               = "test-flow-aggregator-np-egress-deny"
 	ingressAntreaNetworkPolicyName = "test-flow-aggregator-antrea-networkpolicy-ingress"
 	egressAntreaNetworkPolicyName  = "test-flow-aggregator-antrea-networkpolicy-egress"
@@ -430,20 +431,28 @@ func testHelper(t *testing.T, data *TestData, podAIPs, podBIPs, podCIPs, podDIPs
 	// LocalServiceAccess tests the case, where Pod and Service are deployed on the same Node and their flow information is exported as IPFIX flow records.
 	t.Run("LocalServiceAccess", func(t *testing.T) {
 		skipIfProxyDisabled(t, data)
-		if !isIPv6 {
-			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcB.Spec.ClusterIP, isIPv6, true, true, false, false, checkBandwidth)
+		// In dual stack cluster, Service IP can be assigned as different IP family from specified.
+		// In that case, source IP and destination IP will align with IP family of Service IP.
+		// For IPv4-only and IPv6-only cluster, IP family of Service IP will be same as Pod IPs.
+		isServiceIPv6 := net.ParseIP(svcB.Spec.ClusterIP).To4() == nil
+		if isServiceIPv6 {
+			checkRecordsForFlows(t, data, podAIPs.ipv6.String(), svcB.Spec.ClusterIP, isServiceIPv6, true, true, false, false, checkBandwidth)
 		} else {
-			checkRecordsForFlows(t, data, podAIPs.ipv6.String(), svcB.Spec.ClusterIP, isIPv6, true, true, false, false, checkBandwidth)
+			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcB.Spec.ClusterIP, isServiceIPv6, true, true, false, false, checkBandwidth)
 		}
 	})
 
 	// RemoteServiceAccess tests the case, where Pod and Service are deployed on different Nodes and their flow information is exported as IPFIX flow records.
 	t.Run("RemoteServiceAccess", func(t *testing.T) {
 		skipIfProxyDisabled(t, data)
-		if !isIPv6 {
-			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcC.Spec.ClusterIP, isIPv6, false, true, false, false, checkBandwidth)
+		// In dual stack cluster, Service IP can be assigned as different IP family from specified.
+		// In that case, source IP and destination IP will align with IP family of Service IP.
+		// For IPv4-only and IPv6-only cluster, IP family of Service IP will be same as Pod IPs.
+		isServiceIPv6 := net.ParseIP(svcC.Spec.ClusterIP).To4() == nil
+		if isServiceIPv6 {
+			checkRecordsForFlows(t, data, podAIPs.ipv6.String(), svcC.Spec.ClusterIP, isServiceIPv6, false, true, false, false, checkBandwidth)
 		} else {
-			checkRecordsForFlows(t, data, podAIPs.ipv6.String(), svcC.Spec.ClusterIP, isIPv6, false, true, false, false, checkBandwidth)
+			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcC.Spec.ClusterIP, isServiceIPv6, false, true, false, false, checkBandwidth)
 		}
 	})
 }
@@ -679,6 +688,9 @@ func checkPodAndNodeData(t *testing.T, record, srcPod, srcNode, dstPod, dstNode 
 }
 
 func getUnit64FieldFromRecord(t *testing.T, record string, field string) uint64 {
+	if strings.Contains(record, "TEMPLATE SET") {
+		return 0
+	}
 	splitLines := strings.Split(record, "\n")
 	for _, line := range splitLines {
 		if strings.Contains(line, field) {

--- a/test/e2e/framework.go
+++ b/test/e2e/framework.go
@@ -101,7 +101,7 @@ const (
 	busyboxImage        = "projects.registry.vmware.com/library/busybox"
 	nginxImage          = "projects.registry.vmware.com/antrea/nginx"
 	perftoolImage       = "projects.registry.vmware.com/antrea/perftool"
-	ipfixCollectorImage = "projects.registry.vmware.com/antrea/ipfix-collector:v0.5.2"
+	ipfixCollectorImage = "projects.registry.vmware.com/antrea/ipfix-collector:v0.5.3"
 	ipfixCollectorPort  = "4739"
 
 	nginxLBService = "nginx-loadbalancer"

